Assignment6/Array.cs

```
@@ -1,10 +1,117 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 
 namespace Assignment6
 {
-    public class Array<T>
+    public class ArrayCollection<T> : ICollection<T>
     {
-        
+        public int Capacity { get; }
+
+        private List<T> _Items;
+
+        public ArrayCollection(int width)
+        {
+            if(width <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(width), "Must be greater than 0");
+            }
+
+            Capacity = width;
+            _Items = new List<T>(width);
+        }
+
+        public int Count => _Items.Count;
+
+        public bool IsReadOnly => false;
+
+        public void Add(T item)
+        {
+            if (Count == Capacity)
+            {
+                throw new ArgumentOutOfRangeException(nameof(_Items), "Collection is full!");
+            }
+            else if(item is null)
+            {
+                throw new ArgumentNullException(nameof(item), "Item is null");
+            }
+            _Items.Add(item);
+        }
+
+        public void Clear()
+        {
+            _Items.Clear();
+        }
+
+        public bool Contains(T item)
+        {
+            if (_Items.Contains(item))
+            {
+                return true;
+            }
+            
+            throw new ArgumentException(nameof(item), "Item not in list!");

```

Why throw an exception rather than returning false?  There is no point in a return if the only valid return is true. 
Returnining `_Items.Contains(item)` seems like all you need.